src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.jsx

```diff
@@ -1,51 +1,123 @@
 import React, {
+  createRef,
+  useEffect,
   useRef,
+  useState,
 } from 'react';
 import PropTypes from 'prop-types';
 import FormControl from '../form-control/form-control.jsx';
 import Icon from '../icon/icon.jsx';
 import iconCaretDown from '../../svgs/icon-caret-down.svg';
 import iconCaretDownDisabled from '../../svgs/icon-caret-down-disabled.svg';
+import Listbox from '../listbox/listbox.jsx';
+import ListOption from '../list-option/list-option.jsx';
 import TagList from '../tag-list/tag-list.jsx';
 import Tag from '../tag/tag.jsx';
 import styles from './custom-field-input-multiple-choice.css';
 
 function CustomFieldInputMultipleChoice(props) {
-  const refs = props.value.map(() => useRef());
+  const listboxRef = useRef();
+  const [expanded, setExpanded] = useState(false);
+  const [value, setValue] = useState(props.value);
+  const visibleChoices = props.choices.filter(choice => !value.includes(choice));
+  const choicesRefs = visibleChoices.map(() => createRef());
+  const valueRefs = value.map(() => createRef());
   const classContainer = props.readOnly ?
     styles['read-only-container'] :
     styles['read-write-container'];
 
+  function onChoiceRemove(event) {
+    const newValue = value.filter((choice, index) => (
+      valueRefs[index].current !== event.target
+    ));
+    setValue(newValue);
+  }
+
+  function onChoiceSelect(event) {
+    const selectedChoiceIndex = choicesRefs.findIndex(choiceRef => choiceRef === event.target);
+    const selectedChoice = visibleChoices[selectedChoiceIndex];
+    setExpanded(false);
+    setValue([...value, selectedChoice]);
+  }
+
+  function onClick() {
+    setExpanded(true);
+  }
+
+  function onKeyDown(event) {
+    switch (event.key) {
+      case 'Escape':
+        setExpanded(false);
```

Comment:
Do we want to export the `setExpanded` function with `useImperativeHandle` so that parent components can provide a "click-out" functionality for closing the dropdown?

Reply:
I'm not sure yet on that API! Historically, we'll unfortunately want to set an event listener on the body and do some `node.contains` type of thing. That is based on a per-component basis that has a popup. So for now, we're punting so we can exclusively focus on the issue in a future story.